packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -137,22 +135,50 @@ function s(
     if (
         !isUndefined(slotset) &&
         !isUndefined(slotset[slotName]) &&
-        slotset[slotName].length !== 0
+        slotset[slotName].vnodes.length !== 0
     ) {
-        children = slotset[slotName];
+        aChildren = slotset[slotName];
     }
-    const vmBeingRendered = getVMBeingRendered()!;
-    const { renderMode, shadowMode } = vmBeingRendered;
+
+    const { renderMode, shadowMode } = getVMBeingRendered()!;
 
     if (renderMode === RenderMode.Light) {
-        sc(children);
-        return children;
+        // In LightDom, the effective children are returned as is, we will need to
+        // diff the children (patchChildren) using an empty that is always present (to get the parentElement) and
+        // the same in both collection to patch (so it serves as an anchor).
+
+        // Note on why 2: if the same vnode is reused, it may lead to issues when removing because of elm.(1 vnode for 2 elements)
+        const markerStart = t('\u200D');
+        const markerEnd = t('\u200D');
```

Comment:
Why a ZWJ instead of an empty text node? Because it gets rendered in SSR?

Reply:
FWIW I took a look at how empty slots are rendered by a few other frameworks:

- [Svelte](https://svelte.dev/repl/ed7331fe1d3840ada00a16c631dd831a?version=3.46.4): nothing
- [Vue](https://sfc.vuejs.org/#eyJBcHAudnVlIjoiPHNjcmlwdD5cbmltcG9ydCBDb21wb25lbnQgZnJvbSAnLi9Db21wb25lbnQudnVlJ1xuICBcbmV4cG9ydCBkZWZhdWx0IHtcbiAgbmFtZTogJ0FwcCcsXG4gIGNvbXBvbmVudHM6IHtcbiAgICBDb21wb25lbnRcbiAgfVxufVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPENvbXBvbmVudD48L0NvbXBvbmVudD5cbjwvdGVtcGxhdGU+IiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9zZmMudnVlanMub3JnL3Z1ZS5ydW50aW1lLmVzbS1icm93c2VyLmpzXCJcbiAgfVxufSIsIkNvbXBvbmVudC52dWUiOiI8dGVtcGxhdGU+XG48ZGl2IGNsYXNzPWNvbXBvbmVudD5cblx0PHNsb3Q+PC9zbG90PlxuPC9kaXY+XG48L3RlbXBsYXRlPiJ9): two text nodes each containing the empty string
- [Solid](https://playground.solidjs.com/?hash=1773683703&version=1.3.7): nothing

Both Solid and Svelte don't use vdom, so I guess that's how they get away with doing nothing. I wonder why Vue uses two empty text nodes.

![Screen Shot 2022-02-25 at 1 56 17 PM](https://user-images.githubusercontent.com/283842/155808766-a42f1bae-8efb-4a32-8e13-9737a6c64ccf.png)

Reply:
🤣 yeah, I went with the vue approach based on [your other comment ](https://github.com/salesforce/lwc/pull/2716#discussion_r815173806) 👍